File: app/src/main/java/com/blankj/androidutilcode/feature/core/bar/BarStatusSwipeBackActivity.java
package com.blankj.androidutilcode.feature.core.bar;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.LinearLayout;
import android.widget.SeekBar;
import android.widget.TextView;

import com.blankj.androidutilcode.R;
import com.blankj.androidutilcode.UtilsApp;
import com.blankj.androidutilcode.base.BaseActivity;
import com.blankj.utilcode.util.BarUtils;
import com.r0adkll.slidr.Slidr;

import java.util.Random;


/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2017/05/27
 *     desc  : Bar 工具类 Demo
 * </pre>
 */
public class BarStatusSwipeBackActivity extends BaseActivity {

    private Random mRandom;
    private int    mColor;
    private int    mAlpha;

    private LinearLayout llContainer;
    private CheckBox     cbAlpha;
    private TextView     tvStatusAlpha;
    private SeekBar      sbChangeAlpha;
    private Button       btnRandomColor;

    public static void start(Context context) {
        Intent starter = new Intent(context, BarStatusSwipeBackActivity.class);
        context.startActivity(starter);
    }

    @Override
    public void initData(Bundle bundle) {
        mRandom = new Random();
        mColor = ContextCompat.getColor(UtilsApp.getInstance(), R.color.colorPrimary);
        mAlpha = 112;
    }

    @Override
    public int bindLayout() {
        return R.layout.activity_bar_status_swipe_back;
    }

    @Override
    public void initView(Bundle savedInstanceState, View view) {
        Slidr.attach(this);

        llContainer = findViewById(R.id.ll_container);
        cbAlpha = findViewById(R.id.cb_alpha);
        btnRandomColor = findViewById(R.id.btn_random_color);
        tvStatusAlpha = findViewById(R.id.tv_status_alpha);
        sbChangeAlpha = findViewById(R.id.sb_change_alpha);

        cbAlpha.setOnCheckedChangeListener(mCheckedChangeListener);
        btnRandomColor.setOnClickListener(this);
        findViewById(R.id.btn_set_transparent).setOnClickListener(this);
        sbChangeAlpha.setOnSeekBarChangeListener(mColorListener);

        tvStatusAlpha.setText(String.valueOf(mAlpha));

        updateStatusBar();
    }


    @Override
    public void doBusiness() {

    }

    @Override
    public void onWidgetClick(View view) {
        switch (view.getId()) {
            case R.id.btn_random_color:
                mColor = 0xff000000 | mRandom.nextInt(0xffffff);
                updateStatusBar();
                break;
            case R.id.btn_set_transparent:
                sbChangeAlpha.setProgress(0);
                break;
        }
    }

    private SeekBar.OnSeekBarChangeListener mColorListener = new SeekBar.OnSeekBarChangeListener() {
        @Override
        public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
            mAlpha = progress;
            tvStatusAlpha.setText(String.valueOf(mAlpha));
            updateStatusBar();
        }

        @Override
        public void onStartTrackingTouch(SeekBar seekBar) {

        }

        @Override
        public void onStopTrackingTouch(SeekBar seekBar) {

        }
    };

    CompoundButton.OnCheckedChangeListener mCheckedChangeListener = new CompoundButton.OnCheckedChangeListener() {
        @Override
        public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
            if (isChecked) {
                btnRandomColor.setVisibility(View.GONE);
                llContainer.setBackgroundResource(R.drawable.bg_bar);
            } else {
                btnRandomColor.setVisibility(View.VISIBLE);
                llContainer.setBackgroundColor(Color.WHITE);
            }
            updateStatusBar();
        }
    };

    private void updateStatusBar() {
        if (cbAlpha.isChecked()) {
            BarUtils.setStatusBarAlpha(this, mAlpha);
            BarUtils.addMarginTopEqualStatusBarHeight(cbAlpha);
        } else {
            BarUtils.setStatusBarColor(this, mColor, mAlpha);
            BarUtils.addMarginTopEqualStatusBarHeight(cbAlpha);
        }
    }
}

File: utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/bar/BarStatusSwipeBackActivity.kt
package com.blankj.utilcode.pkg.feature.bar

import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.CompoundButton
import android.widget.SeekBar
import com.blankj.lib.base.BaseActivity
import com.blankj.utilcode.pkg.R
import com.blankj.utilcode.util.BarUtils
import com.blankj.utilcode.util.ColorUtils
import com.r0adkll.slidr.Slidr
import kotlinx.android.synthetic.main.activity_bar_status_swipe_back.*

/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2017/05/27
 * desc  : demo about BarUtils
 * ```
 */
class BarStatusSwipeBackActivity : BaseActivity() {

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, BarStatusSwipeBackActivity::class.java)
            context.startActivity(starter)
        }
    }

    private var mColor: Int = 0
    private var mAlpha: Int = 0

    private val mColorListener = object : SeekBar.OnSeekBarChangeListener {
        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
            mAlpha = progress
            barStatusSwipeBackAboutTv.text = mAlpha.toString()
            updateStatusBar()
        }

        override fun onStartTrackingTouch(seekBar: SeekBar) {}

        override fun onStopTrackingTouch(seekBar: SeekBar) {}
    }

    private var mCheckedChangeListener: CompoundButton.OnCheckedChangeListener = CompoundButton.OnCheckedChangeListener { buttonView, isChecked ->
        if (isChecked) {
            barStatusSwipeBackChangeAlphaSb.visibility = View.VISIBLE
            barStatusSwipeBackRandomColorBtn.visibility = View.GONE
            barStatusSwipeBackSetTransparentBtn.visibility = View.VISIBLE

            barStatusSwipeBackContainerLl.setBackgroundResource(R.drawable.bar_status_alpha_bg)
        } else {
            barStatusSwipeBackChangeAlphaSb.visibility = View.GONE
            barStatusSwipeBackRandomColorBtn.visibility = View.VISIBLE
            barStatusSwipeBackSetTransparentBtn.visibility = View.GONE

            barStatusSwipeBackContainerLl.setBackgroundColor(Color.WHITE)
        }
        updateStatusBar()
    }

    override fun isSwipeBack(): Boolean {
        return true
    }

    override fun initData(bundle: Bundle?) {
        mColor = ColorUtils.getColor(R.color.colorPrimary)
        mAlpha = 112
    }

    override fun bindLayout(): Int {
        return R.layout.activity_bar_status_swipe_back
    }

    override fun initView(savedInstanceState: Bundle?, contentView: View) {
        Slidr.attach(this)
        barStatusSwipeBackAlphaCb.setOnCheckedChangeListener(mCheckedChangeListener)
        barStatusSwipeBackChangeAlphaSb.setOnSeekBarChangeListener(mColorListener)
        barStatusSwipeBackRandomColorBtn.setOnClickListener(this)
        barStatusSwipeBackSetTransparentBtn.setOnClickListener(this)

        barStatusSwipeBackSetTransparentBtn.visibility = View.GONE
        updateStatusBar()
    }


    override fun doBusiness() {}

    override fun onWidgetClick(view: View) {
        when (view.id) {
            R.id.barStatusSwipeBackRandomColorBtn -> {
                mColor = ColorUtils.getRandomColor()
                updateStatusBar()
            }
            R.id.barStatusSwipeBackSetTransparentBtn -> barStatusSwipeBackChangeAlphaSb.progress = 0
        }
    }

    private fun updateStatusBar() {
        if (barStatusSwipeBackAlphaCb.isChecked) {
            BarUtils.setStatusBarColor(this, Color.argb(mAlpha, 0, 0, 0))
            barStatusSwipeBackAboutTv.text = mAlpha.toString()
        } else {
            BarUtils.setStatusBarColor(this, mColor)
            barStatusSwipeBackAboutTv.text = ColorUtils.int2ArgbString(mColor)
        }
        BarUtils.addMarginTopEqualStatusBarHeight(barStatusSwipeBackAlphaCb)
    }
}
